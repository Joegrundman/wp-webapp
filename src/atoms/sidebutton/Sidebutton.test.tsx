import { shallow, ShallowWrapper } from 'enzyme'
import * as React from 'react'
import SideButton from './Sidebutton'

describe('<SideButton/>', () => {
  const defaultProps = {
    action: jest.fn(),
    image: 'mockImg',
  }

  it('should render as expected with default props', () => {
    const component: ShallowWrapper = shallow(<SideButton {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })

  it('should have an action that is triggered when clicked', () => {
    const component: ShallowWrapper = shallow(<SideButton {...defaultProps} />)
    component.find('button').first().simulate('click')
    expect(defaultProps.action).toHaveBeenCalled();
  })
})
