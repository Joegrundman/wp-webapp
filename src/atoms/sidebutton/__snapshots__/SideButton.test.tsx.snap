// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideButton/> should render as expected with default props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebutton
    action={[MockFunction]}
    image="mockImg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "onClick": [MockFunction],
      "style": Object {
        "background": "url(mockImg)",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "onClick": [MockFunction],
        "style": Object {
          "background": "url(mockImg)",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
